src/main/fox_demo.c

```diff
@@ -0,0 +1,3005 @@
+                }
+                arg0->unk_1D0 = 3;
+                func_800A5FA0(&arg0->unk_460, 0x09000002U, arg0->num);
+                arg0->unk_194 = 7.0f;
+                arg0->unk_190 = 7.0f;
+            }
+            arg0->unk_004 -= 0.2f;
+            break;
+        case 3:
+            Math_SmoothStepToF(&arg0->unk_12C, 0.0f, 0.05f, 5.0f, 0.0f);
+            sp58.y = arg0->unk_12C * 0.3f;
+            Math_SmoothStepToF(&arg0->wings.unk_04, sp58.y, 0.3f, 100.0f, 0.0f);
+            Math_SmoothStepToF(&arg0->wings.unk_08, sp58.y, 0.3f, 100.0f, 0.0f);
+            sp58.z = -sp58.y;
+            Math_SmoothStepToF(&arg0->wings.unk_0C, sp58.z, 0.3f, 100.0f, 0.0f);
+            Math_SmoothStepToF(&arg0->wings.unk_10, sp58.z, 0.3f, 100.0f, 0.0f);
+            arg0->unk_190 = 2.0f;
+            arg0->camEye.x += arg0->vel.x * 0.1f;
+            arg0->camEye.z += arg0->vel.z * 0.1f;
+            if (arg0->unk_19C != 0) {
+                arg0->camEye.x += arg0->vel.z * 0.2f;
+                arg0->camEye.z += arg0->vel.x * 0.2f;
+            }
+            if (arg0->unk_19C == 0) {
+                Math_SmoothStepToAngle(&arg0->unk_114, sp58.x, 0.1f, 2.0f, 0.0f);
+            }
+            if (arg0->pos.y < arg0->unk_0A0) {
+                if (arg0->unk_004 < 0.0f) {
+                    arg0->unk_004 += 0.2f;
+                }
+            } else {
+                arg0->unk_004 -= 0.2f;
+            }
+            if (arg0->timer_1F8 == 0) {
+                arg0->state_1C8 = PLAYERSTATE_1C8_3;
+                arg0->unk_014 = 0.0f;
+                arg0->unk_018 = 0.0f;
+            }
+            break;
+    }
+    arg0->pos.y += arg0->unk_004;
+    arg0->camAt.y += arg0->unk_004;
+    arg0->unk_0F8 = arg0->unk_0EC + arg0->unk_12C + arg0->unk_130;
+    Matrix_RotateY(gCalcMatrix, (arg0->unk_114 + arg0->unk_0E8 + 180.0f) * M_DTOR, 0);
+    Matrix_RotateX(gCalcMatrix, -((arg0->unk_120 + arg0->unk_0E4 + arg0->unk_4D8) * M_DTOR), 1);
+    sp50.rot.x = 0.0f;
+    sp50.rot.y = 0.0f;
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp50.rot, &sp50.pos);
+    arg0->vel.x = sp50.pos.x;
+    arg0->vel.z = sp50.pos.z;
+    arg0->vel.y = sp50.pos.y;
+    arg0->pos.x += arg0->vel.x;
+    arg0->pos.y += arg0->vel.y;
+    if (arg0->pos.y < arg0->unk_0A4) {
+        arg0->pos.y = arg0->unk_0A4;
+        arg0->vel.y = 0.0f;
+    }
+    arg0->pos.z += arg0->vel.z;
+    arg0->unk_138 = arg0->pos.z;
+    func_800B2574(arg0);
+    func_800B2130(arg0);
+    func_800AD7F0(arg0);
+    func_800AA800(arg0);
+    func_800A46A0(arg0);
+}
+
+void func_8004D3C8(s32 arg0) {
+}
+
+void func_8004D3D4(Player* player, s32 arg1, Item* arg2) {
+    u16 itemId;
+
+    Item_Initialize(arg2);
+
+    arg2->obj.status = 1;
+    arg2->obj.pos.x = player->pos.x;
+    arg2->obj.pos.y = player->pos.y;
+    arg2->obj.pos.z = player->unk_138;
+    itemId = arg1 & 0xFFFF;
+    arg2->obj.id = itemId;
+    Object_SetInfo(&arg2->info, itemId);
```

Comment:
```suggestion
    arg2->obj.id = itemId;
    Object_SetInfo(&arg2->info, arg2->obj.id);
```
Does this work? Also arg2 should be named `item`.